### PR TITLE
feat: Finding no. of digits in a Number

### PR DESCRIPTION
**Pull request by amino19**
#### Description of Change

The changes I made here, to commit the code for finding the no. of digits in a Number.
Ref. to some close PR #1491 & #1492   

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:  #1491 & #1492 are closed.. This can be treat as new commit, I delete their remote also, so the changes can only be done to the right folder, nothing will be messed up..

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1497"></a>

### CODE REVIEW

File: math/finding_number_of_digits_in_a_number.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
+ * @author amino19
+ * @file
+ *
+ * \brief Finding number of Digits in a Number
```
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:5]: Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

File: math/finding_number_of_digits_in_a_number.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *  For more details, refer Algorithms-Explanation
+ */
+
+/* Log based C++ program to count number of
+ *  digits in a number/integer */
+
+#include <cassert>
+#include <iostream>
+
+int main() {
```
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:17]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main() {
```

File: math/finding_number_of_digits_in_a_number.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/**
+ * @author amino19
+ * @file
+ *
+ * \brief Finding number of Digits in a Number
+ * \details Basically, its a very basic math of
```
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:6]: You should provide a better-detailed description (see, for example, the [`n_queens.cpp`](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/backtracking/n_queens.cpp) algorithm).

File: math/finding_number_of_digits_in_a_number.cpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ *  finding number of digits in a number (input),
+ *  For more details, refer Algorithms-Explanation
+ */
+
+/* Log based C++ program to count number of
+ *  digits in a number/integer */
+
+#include <cassert>
+#include <iostream>
```
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:15]:
```suggestion
#include <cassert>      /// for assert
#include <iostream>    /// for IO operations
```

File: math/finding_number_of_digits_in_a_number.cpp [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+     *  remove last digit from n in each iteration
+     *  increase count by 1 in each iteration */
+
+    while (n != 0) {
+        n /= 10;  // n = n/10
+        ++count;
+    }
+
+    std::cout << "Number of digits: " << count;
+}
```
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:33]:
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Good work.

File: math/finding_number_of_digits_in_a_number.cpp (hunk 1 above)
Comment (amino19) [reply math/finding_number_of_digits_in_a_number.cpp:5]: Ya, already attached a webpage in Explanation part. 
If that needs more changes, pl do acknowledge me!

Comment (Panquesito7) [reply math/finding_number_of_digits_in_a_number.cpp]: It needs to be added here, too. It's needed for the Doxygen documentation.

File: math/finding_number_of_digits_in_a_number.cpp (hunk 3 above)
Comment (amino19) [reply math/finding_number_of_digits_in_a_number.cpp:6]: Actually its a very short code, but I'll try my best to figure it out!

File: math/finding_number_of_digits_in_a_number.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
+/**
+ * @author amino19
```
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:2]:
```suggestion
 * @author [ANSHUMAAN](https://github.com/amino19)
```

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
+/**
+ * @author [ANSHUMAAN](https://github.com/amino19)
+ * @file:[Program to count digits in an
+ * integer](https://www.geeksforgeeks.org/program-count-digits-integer-3-different-methods)
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:4]:
```suggestion
 * @file [Program to count digits in an
 * integer](https://www.geeksforgeeks.org/program-count-digits-integer-3-different-methods) in 3 different methods
```

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    while (n != 0) {
+        /* we can also use: n = n/10 */
+        n /= 10;
+        /* each time while loop running, count will
+         *  be increment by 1.
+         */
+        ++count;
+    }
+    std::cout << "Number of digits: " << count;
+}
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:42]:
```suggestion
}

```

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+ * \details It is a very basic math of finding
+ *  number of digits in a given number i.e, we
+ *  can use it by inputting values it can be a
+ *  positive/negatie value, lets say integer.
+ *
+ *  For more details, refer Algorithms-Explanation
+ */
+
+/* Log based C++ program to count number of
+ *  digits in a number/integer */
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:16]:
```suggestion
```

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
+/**
+ * @author [ANSHUMAAN](https://github.com/amino19)
+ * @file:[Program to count digits in an
+ * integer](https://www.geeksforgeeks.org/program-count-digits-integer-3-different-methods)
+ *
+ * \brief Finding number of Digits in a Number
+ * \details It is a very basic math of finding
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:7]: Please consider adding a better-detailed description, as I previously said.

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        /*  each time while loop running, count will
+         *  be increment by 1.
+         */
+        ++count;
+    }
+
+    std::cout << "Number of digits: " << count;
+
+    return 0;
+}
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:48]: Please accept this suggestion.
```suggestion
}

```

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    int n = 0;
+    int count = 0;
+
+    /* 'cout' to ask input from user..
+     * 'cin' to taking integer as input..
+     */
+    std::cout << "Enter an integer: ";
+    std::cin >> n;
+
+    /* iterate until n becomes 0
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:33]: Preferably, use one-line comments (`//`) inside functions. We use multi-line comments outside functions.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Good work.

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+ * @file [Program to count digits in an
+ * integer](https://www.geeksforgeeks.org/program-count-digits-integer-3-different-methods)
+ *
+ * @brief Finding number of Digits in a Number
+ * @details It is a very basic math of finding number of digits in a given
+ * number i.e, we can use it by inputting values whether it can be a
+ * positive/negative value, lets say integer. There is also second method to do
+ * it with, by using "K = floor(log10(N) + 1)", but its only applicable for
+ * numbers (not integers).
+ * For more details, refer Algorithms-Explanation
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:12]: Once your PR https://github.com/TheAlgorithms/Algorithms-Explanation/pull/101 is merged, this link will work properly.
```suggestion
 * For more details, refer to the [Algorithms-Explanation](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/en/Basic%20Math/Finding the number of digits in a number.md) repository.
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Comment by amino19** [comment]
@Panquesito7 : Done with all changes!

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+ * @returns 0 on exit
+ */
+int main() {
+    // Initialize `n` & `count` by 0
+    int n = 0;
+    int count = 0;
+
+    /* 'cout' to ask input from user..
+     * 'cin' to taking integer as input..
+     */
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:29]: > Make sure you put useful comments in your code. Do not comment on things that are obvious.

```suggestion
```

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+        // we can also use: n = n/10
+        n /= 10;
+        /*  each time while loop running, count will
+         *  be increment by 1.
+         */
+        ++count;
+    }
+
+    std::cout << "Number of digits: " << count;
+
```
[the remark this answers is not in the document]
Comment (Panquesito7) [reply math/finding_number_of_Digits_in_a_Number.cpp:43]: Please accept this suggestion.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/finding_number_of_Digits_in_a_Number.cpp (hunk 14 above)
Comment (amino19) [reply math/finding_number_of_Digits_in_a_Number.cpp]: Yeah, I think you might need perfects basic comments in almost all line!
BDW, just drop perfect name my perfect name! :sweat_smile:

File: math/finding_number_of_Digits_in_a_Number.cpp [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+ * number i.e, we can use it by inputting values whether it can be a
+ * positive/negative value, let's say: an integer. There is also a second method: 
+ * by using "K = floor(log10(N) + 1)", but it's only applicable for
+ * numbers (not integers).
+ * For more details, refer to the [Algorithms-Explanation](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/en/Basic%20Math/Finding the number of digits in a number.md) repository.
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+
```
Comment (Panquesito7) [inline math/finding_number_of_Digits_in_a_Number.cpp:17]: Please add self-test implementations using `assert` (see the [typical structure](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) of a program as an example).

**Review by Panquesito7 — COMMENTED** [review]
Almost there. 😄

File: math/finding_number_of_Digits_in_a_Number.cpp (hunk 16 above)
Comment (amino19) [reply math/finding_number_of_Digits_in_a_Number.cpp:17]: @Panquesito7 : Help me for this! :smile:

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Awesome work, @amino19! Thanks for your contribution! 😄👍🎉
*Note for maintainers: the other files are modified automatically by the CI.*

**Review by mishraabhinn — APPROVED** [review]

**Pull request merged**